File: CS-Examples/14_Hyperlinks/RemoveHyperlinks.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Collections;

namespace RemoveHyperlinks
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\HyperlinksSample1.xlsx");

            //Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            //Get the collection of all hyperlinks in the worksheet
            HyperLinksCollection links = sheet.HyperLinks;

            // Remove the content of cells B1, B2, B3 to remove link text
            sheet.Range["B1"].ClearAll();
            sheet.Range["B2"].ClearAll();
            sheet.Range["B3"].ClearAll();

            // Remove hyperlink
            sheet.HyperLinks.RemoveAt(0);
            sheet.HyperLinks.RemoveAt(0);
            sheet.HyperLinks.RemoveAt(0);

            // Specify the output file name for the modified workbook
            string output = "RemoveHyperlinks.xlsx";

            // Save the modified workbook to the specified file using Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/14_Hyperlinks/RemoveHyperlinks.vb
Imports Spire.Xls
Imports Spire.Xls.Collections

Namespace RemoveHyperlinks

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a workbook
			Dim workbook As New Workbook()

			'Load the document from disk
			workbook.LoadFromFile("..\..\..\..\..\..\Data\HyperlinksSample1.xlsx")

			'Get the first worksheet
			Dim sheet As Worksheet = workbook.Worksheets(0)

			'Get the collection of all hyperlinks in the worksheet
			Dim links As HyperLinksCollection = sheet.HyperLinks

			'Remove all link content
			sheet.Range("B1").ClearAll()
			sheet.Range("B2").ClearAll()
			sheet.Range("B3").ClearAll()

			'Remove hyperlink and keep link text
			sheet.HyperLinks.RemoveAt(0)
			sheet.HyperLinks.RemoveAt(0)
			sheet.HyperLinks.RemoveAt(0)

			'Save the document
			Dim output As String = "RemoveHyperlinks.xlsx"
			workbook.SaveToFile(output, ExcelVersion.Version2013)

			'Launch the Excel file
			ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
